addons/hr_attendance/models/hr_employee.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models, _
+from openerp.exceptions import UserError
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+    _description = "Employee"
+
+    state = fields.Selection([('absent', 'Absent'), ('present', 'Present')], compute='_state', string='Attendance')
+    last_sign = fields.Datetime(compute='_last_sign', string='Last Sign')
+    attendance_access = fields.Boolean(compute='_attendance_access', string='Attendance Access')
+
+    @api.multi
+    def _state(self):
+        for emp in self:
+            emp.state = 'absent'
+        self.env.cr.execute('SELECT hr_attendance.action, hr_attendance.employee_id \
+                FROM ( \
+                    SELECT MAX(name) AS name, employee_id \
+                    FROM hr_attendance \
+                    WHERE action in (\'sign_in\', \'sign_out\') \
+                    GROUP BY employee_id \
+                ) AS foo \
+                LEFT JOIN hr_attendance \
+                    ON (hr_attendance.employee_id = foo.employee_id \
+                        AND hr_attendance.name = foo.name) \
+                WHERE hr_attendance.employee_id IN %s', (tuple(self.ids),))
+        for res in self.env.cr.fetchall():
+            emp.state = res[0] == 'sign_in' and 'present' or 'absent'
```

Comment:
This is confusing, you are accessing emp of previous loop.
self.env.cr.fetchall() will return [('sign_in', 1), ('sign_out', 2)] and your self is recordset, so you need to assign action(sign_in/sign_out) to respective record